pypeit/edgetrace.py

```diff
@@ -960,384 +976,486 @@ def initial_trace(self, bpm=None, save=False):
         # Restart the log
         self.log = [inspect.stack()[0][3]]
 
-        # Save if requested
-        if save:
-            self.save()
-
-    def save(self, outfile, overwrite=True, checksum=True, float_dtype='float32',
-             master_dir=None, master_key=None):
+    def _base_header(self, hdr=None):
         """
-        Save the trace object for later re-instantiation.
-
-        Args:
-            outfile (:obj:`str`):
-                Full path for the output file. Defaults to
-                :attr:`master_file_path`.
-            overwrite (:obj:`bool`, optional):
-                Overwrite any existing file.
-            checksum (:obj:`bool`, optional):
-                Passed to `astropy.io.fits.HDUList.writeto`_ to add
-                the DATASUM and CHECKSUM keywords fits header(s).
-            float_dtype (:obj:`str`, optional):
-                Convert floating-point data to this data type before
-                writing.  Default is 32-bit precision.
-        """
-
-        #_outfile = self.master_file_path if outfile is None else outfile
+        Construct the baseline header for all HDU extensions.
 
-        # Check if it exists
-        if os.path.exists(outfile) and not overwrite:
-            msgs.error('Master file exists: {0}'.format(outfile) + msgs.newline()
-                       + 'Set overwrite=True to overwrite it.')
+        This appends the :class:`EdgeTracePar` and
+        :class:`EdgeTraceBitMask` data to the headers of all HDU
+        extensions. This is overkill, but avoids overriding
```

Comment:
I wonder if this can be applied to a few other existing `DataContainer`'s ?

Reply:
Yep, should be easy enough.  That's mainly why I added the two header methods to DataContainer.  And the `par.to_header` and `bitmask.to_header` methods are generic to the `ParSet` and `BitMask` classes.  Although I think `ParSet.to_header` will barf if there's a nested `ParSet`.